fastly/acl_entry.go

```diff
@@ -63,6 +71,112 @@ func (c *Client) ListACLEntries(i *ListACLEntriesInput) ([]*ACLEntry, error) {
 	return es, nil
 }
 
+type ListAclEntriesPaginator struct {
+	consumed    bool
+	CurrentPage int
+	NextPage    int
+	LastPage    int
+	client      *Client
+	options     *ListACLEntriesInput
+}
+
+// HasNext returns a boolean indicating whether more pages are available
+func (p *ListAclEntriesPaginator) HasNext() bool {
+	return !p.consumed || p.Remaining() != 0
+}
+
+// Remaining returns the remaining page count
+func (p *ListAclEntriesPaginator) Remaining() int {
+	if p.LastPage == 0 {
+		return 0
+	}
+	return p.LastPage - p.CurrentPage
+}
+
+// GetNext retrieves data in the next page
+func (p *ListAclEntriesPaginator) GetNext() ([]*ACLEntry, error) {
+	return p.client.listACLEntriesWithPage(p.options, p)
+}
+
+// NewListACLEntriesPaginator returns a new ListAclEntriesPaginator
+func (c *Client) NewListACLEntriesPaginator(i *ListACLEntriesInput) *ListAclEntriesPaginator {
+	return &ListAclEntriesPaginator{
+		client:  c,
+		options: i,
+	}
+}
+
+// listACLEntriesWithPage return a list of entries for an ACL of a given page
+func (c *Client) listACLEntriesWithPage(i *ListACLEntriesInput, p *ListAclEntriesPaginator) ([]*ACLEntry, error) {
+
+	if i.ServiceID == "" {
+		return nil, ErrMissingServiceID
+	}
+
+	if i.ACLID == "" {
+		return nil, ErrMissingACLID
+	}
+
+	var perPage int
+	if i.PerPage <= 0 {
+		// Max is 100
+		perPage = 100
+	} else {
+		perPage = i.PerPage
+	}
+
+	if i.Page <= 0 && p.CurrentPage == 0 {
+		p.CurrentPage = 1
+	} else {
+		p.CurrentPage = p.CurrentPage + 1
+	}
+
+	path := fmt.Sprintf("/service/%s/acl/%s/entries", i.ServiceID, i.ACLID)
+	requestOptions := &RequestOptions{
+		Params: map[string]string{
+			"per_page": strconv.Itoa(perPage),
+			"page":     strconv.Itoa(p.CurrentPage),
+		},
+	}
+
+	if i.Direction != "" {
+		requestOptions.Params["direction"] = i.Direction
+	}
+	if i.Sort != "" {
+		requestOptions.Params["sort"] = i.Sort
+	}
+
+	resp, err := c.Get(path, requestOptions)
+	if err == nil {
```

Comment:
I think we should replace the if/else with a single if conditional (as it aligns more closely with the traditional error handling pattern and avoids needing an `else` block)...

```go
if err != nil {
    return nil, err
}

for _, l := range link.ParseResponse(resp) {
    // ...
}

p.consumed = true
```